### DISCUSSION TITLE
Relationship: Adjacency List or alternative?

### DISCUSSION THREAD

**Discussion by buhtz** [sqlalchemy/sqlalchemy#6351] [Usage Questions] [answered]
Using `sqlalchemy 1.4.9` with `Python 3.9.2` on a Windows 10 64 bit.

I am sorry for asking. Because your documentation is great and awsome.
And I am sure the solution to my question is somewhere in your docu. But I am a bit confused in my head about my situation. Of course I know https://docs.sqlalchemy.org/en/14/orm/self_referential.html and https://docs.sqlalchemy.org/en/14/orm/relationship_api.html#sqlalchemy.orm.relationship

I am not the constructor of the underlying sqllite database. But I just have to deal with it.

Problem in short: I have a Table/Class that can have one parent and multiple children of the same Table/Class.

That are the two relevant tables described in SQLlite SQL

```
CREATE TABLE Reference (
    ID GUID CONSTRAINT PK_Reference PRIMARY KEY,
    ...
)

CREATE TABLE ReferenceReference (
    ParentReferenceID GUID, 
    ChildReferenceID GUID, 
    PRIMARY KEY (ParentReferenceID, 
    ChildReferenceID)
)

```

That is my current Python3 representation of it

```
class Reference(_Base):
    __tablename__ = 'Reference'
    _id = sa.Column('ID', UUIDType, primary_key=True)

ReferenceReference = sa.Table('ReferenceReference', _Base.metadata,
        sa.Column('ParentReferenceID', UUIDType, sa.ForeignKey('Reference.ID'), primary_key=True),
        sa.Column('ChildReferenceID', UUIDType, sa.ForeignKey('Reference.ID'), primary_key=True)
)
```

What I want is to access the `parent` and the `children` of a `Reference` object/row. So I need something like this

```
class Reference(_Base):
    _parent = ...
    _children = ...
```

The first example on [Adjacency List Relationships](https://docs.sqlalchemy.org/en/14/orm/self_referential.html) is realy near my needs. But it does not have a linking table like `ReferenceReference` (what whould be `NodeNode` in the documentations example).

**Comment by zzzeek** [comment, marked as answer]
hey there -

I think we should add a "seealso" to that section.  for your tables you want the many-to-many version at https://docs.sqlalchemy.org/en/14/orm/join_conditions.html#self-referential-many-to-many-relationship .

**Comment by buhtz** [comment]
Thanks for the suggestion. It work as expected with that code

```
class Reference(Base):
    _parent = sao.relationship('Reference',
                               secondary=ReferenceReference,
                               primaryjoin=_id==ReferenceReference.c.ChildReferenceID,
                               secondaryjoin=_id==ReferenceReference.c.ParentReferenceID,
                               backref="_children")
```
